dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -0,0 +1,206 @@
+
+  // what was previously used so we other wallets could find funds
+  //  private val signingKeyHDAddress = {
+//    val coin = HDCoin(HDPurposes.SegWit, HDCoinType.fromNetwork(conf.network))
+//    val account = HDAccount(coin, 0)
+//    val chain = HDChain(HDChainType.External, account)
+//    HDAddress(chain, 0)
+//  }
+
+  // 585 is a random one I picked, unclaimed in https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+  private val PURPOSE = 585
+
+  private val rValueAccount: HDAccount = {
+    val coin = HDCoin(HDPurpose(PURPOSE), HDCoinType.fromNetwork(conf.network))
+    HDAccount(coin, 0)
+  }
+
+  private val chainIndex = 0
+
+  private val signingKey: ECPrivateKey = {
+    val accountIndex = rValueAccount.index
+    val coin = rValueAccount.coin
+    val purpose = coin.purpose
+    // Use a different chain index so we don't reuse a key
+    val chain = chainIndex + 1
+
+    val path = BIP32Path.fromString(
+      s"m/${purpose.constant}'/${coin.coinType.toInt}'/$accountIndex'/$chain'/0'")
+
+    // what was previously used so we other wallets could find funds
+//    extPrivateKey
+//      .deriveChildPrivKey(SegWitHDPath(signingKeyHDAddress))
+//      .key
+
+    extPrivateKey.deriveChildPrivKey(path).key
+  }
+
+  val publicKey: SchnorrPublicKey = signingKey.schnorrPublicKey
+
+  def stakingAddress(network: BitcoinNetwork): Bech32Address =
+    Bech32Address(P2WPKHWitnessSPKV0(signingKey.publicKey), network)
+
+  protected[bitcoins] val rValueDAO: RValueDAO = RValueDAO()
+  protected[bitcoins] val eventDAO: EventDAO = EventDAO()
+  protected[bitcoins] val eventOutcomeDAO: EventOutcomeDAO = EventOutcomeDAO()
+
+  private def getPath(keyIndex: Int): BIP32Path = {
+    val accountIndex = rValueAccount.index
+    val coin = rValueAccount.coin
+    val purpose = coin.purpose
+
+    BIP32Path.fromString(
+      s"m/${purpose.constant}'/${coin.coinType.toInt}'/$accountIndex'/$chainIndex'/$keyIndex'")
+  }
+
+  private def getKValue(rValDb: RValueDb): ECPrivateKey =
+    getKValue(rValDb.label, rValDb.path)
+
+  private def getKValue(label: String, path: BIP32Path): ECPrivateKey = {
+    require(path.forall(_.hardened),
+            s"Cannot use a BIP32Path with unhardened nodes, got $path")
+    val priv = extPrivateKey.deriveChildPrivKey(path).key
+    val hash =
+      CryptoUtil.sha256(
+        priv.schnorrNonce.bytes ++ CryptoUtil.serializeForHash(label))
+    val tweak = ECPrivateKey(hash.bytes)
+
+    priv.add(tweak)
+  }
+
+  def listEventDbs(): Future[Vector[EventDb]] = eventDAO.findAll()
+
+  def listPendingEventDbs(): Future[Vector[EventDb]] = eventDAO.getPendingEvents
+
+  def listEvents(): Future[Vector[Event]] = {
+    for {
+      eventDbs <- eventDAO.findAll()
+      outcomes <- eventOutcomeDAO.findAll()
+    } yield {
+      val outcomesByNonce = outcomes.groupBy(_.nonce)
+      eventDbs.map(db => Event(db, outcomesByNonce(db.nonce)))
+    }
+  }
+
+  def createNewEvent(
+      label: String,
```

Comment:
I'm not sure where you want to address this, but in the future we will need to take a `pubKey` and `expiration` as part of #2099 and #2101 . I understand that you don't think it's wise to provide the abililty to rotate keys, but just also from the oracle explorer perspective it makes it _much_ easier to just store the pubkey in the database so I can just read it out of the database instead of having to touch private key material to get the public key.

Reply:
Imo I think it makes more sense for the dlc explorer to use a different db schema than a oracle implementation

Reply:
>Imo I think it makes more sense for the dlc explorer to use a different db schema than a oracle implementation

This may be true, but it still doesn't address the problem of having to have access to the private key to serve oracle information. I think it's a bad idea to require access to the private key to give people an oracle address for instance. 

Also I think we both agree expirations are a good idea in the oracle database ?